ciscosma_connector.py

```diff
@@ -0,0 +1,1165 @@
+
+                if response.status_code != 200:
+                    return (
+                        action_result.set_status(
+                            phantom.APP_ERROR, f"API call failed. Status code: {response.status_code}. Response: {response.text}"
+                        ),
+                        None,
+                    )
+
+                return phantom.APP_SUCCESS, resp_json
+            else:
+                # Return raw response object for non-JSON responses (downloading attachment)
+                if response.status_code != 200:
+                    return (
+                        action_result.set_status(
+                            phantom.APP_ERROR, f"API call failed. Status code: {response.status_code}. Response: {response.text}"
+                        ),
+                        None,
+                    )
+
+                return phantom.APP_SUCCESS, response
+
+        except requests.exceptions.RequestException as e:
+            return action_result.set_status(phantom.APP_ERROR, f"Error connecting to server: {str(e)}"), None
+        except Exception as e:
+            return action_result.set_status(phantom.APP_ERROR, f"Error making REST call: {str(e)}"), None
+
+    def _make_authenticated_request(self, action_result, endpoint, headers=None, params=None, data=None, json_data=None, method="get"):
+        """Function that makes authenticated REST calls to the app with auth details.
+
+        :param endpoint: REST endpoint that needs to be appended to the service address
+        :param action_result: object of ActionResult class
+        :param headers: request headers
+        :param params: request parameters
+        :param data: request body
+        :param json_data: JSON object
+        :param method: GET/POST/PUT/DELETE/PATCH (Default will be GET)
+        :return: status phantom.APP_ERROR/phantom.APP_SUCCESS(along with appropriate message),
+        response obtained by making an API call
+        """
+        url = f"{self._base_url}{endpoint}"
+        if headers is None:
+            headers = {"Content-Type": "application/json", "Accept": "application/json"}
+
+        # Basic Auth header with encoded credentials
+        basic_auth = base64.b64encode(f"{self._username}:{self._password}".encode()).decode()
+        headers.update({"Authorization": f"Basic {basic_auth}"})
+
+        ret_val, resp_json = self._make_rest_call(url, action_result, headers, params, data, json_data, method)
+
+        if phantom.is_fail(ret_val):
+            return action_result.get_status(), None
+
+        return phantom.APP_SUCCESS, resp_json
+
+    def _sanitize_file_name(self, file_name):
+        """Helper to sanitize file name.
+
+        Args:
+            file_name (str): The file name to sanitize
+
+        Returns:
+            str: Sanitized file name
+        """
+        return re.sub("[,\"']", "", file_name)
```

Comment:
Nit: strongly prefer using "raw" strings for regex, if only because most IDEs highlight them better. Also, I usually prefer to use docstring syntax over quote escaping when I can:
```suggestion
        return re.sub(r"""[,"']""", "", file_name)
```

Reply:
Done 👍